src/components/ProjectSettingsTabs/Header.tsx

```diff
@@ -0,0 +1,141 @@
+import { useMutation } from "@apollo/client";
+import styled from "@emotion/styled";
+import { H2, Disclaimer } from "@leafygreen-ui/typography";
+import { Button } from "components/Button";
+import { ProjectSettingsTabRoutes } from "constants/routes";
+import { useToastContext } from "context/toast";
+import {
+  ProjectSettingsSection,
+  SaveProjectSettingsForSectionMutation,
+  SaveProjectSettingsForSectionMutationVariables,
+  SaveRepoSettingsForSectionMutation,
+  SaveRepoSettingsForSectionMutationVariables,
+} from "gql/generated/types";
+import {
+  SAVE_PROJECT_SETTINGS_FOR_SECTION,
+  SAVE_REPO_SETTINGS_FOR_SECTION,
+} from "gql/mutations";
+import { getTabTitle } from "pages/projectSettings/getTabTitle";
+import { useIsTabSaved, useProjectSettingsContext } from "./Context";
+import { TransformerMap, WritableTabRoutes } from "./types";
+
+interface Props {
+  id: string;
+  isRepo: boolean;
+  saveable: boolean;
+  tab: ProjectSettingsTabRoutes;
+  useRepoSettings: boolean;
+}
+
+export const Header: React.FC<Props> = ({
+  id,
+  isRepo,
+  saveable,
+  tab,
+  useRepoSettings,
+}) => {
+  const dispatchToast = useToastContext();
+  const { title, subtitle } = getTabTitle(tab);
+  const { getTab, saveTab } = useProjectSettingsContext();
+  const { hasError } = getTab(tab);
+  const saved = useIsTabSaved(tab);
+
+  const [saveProjectSection] = useMutation<
+    SaveProjectSettingsForSectionMutation,
+    SaveProjectSettingsForSectionMutationVariables
+  >(SAVE_PROJECT_SETTINGS_FOR_SECTION, {
+    onCompleted() {
+      dispatchToast.success("Successfully updated project");
+    },
+    onError(err) {
+      dispatchToast.error(
+        `There was an error saving the project: ${err.message}`
+      );
+    },
+    refetchQueries: ["ProjectSettings"],
```

Comment:
We can avoid `refetchQueries` if we make the mutation return the updated form state. Since the `saveProjectSettingsForSection` mutation returns the `projectSettings` object we can just have it return all the fields that were updated in the same request.

Reply:
Apollo will handle  updating the `ProjectSettings` response with the data returned from the mutation automatically. So long as there is a unique id identifying the responses.